src/common/transformations/include/transformations/op_conversions/reduce_modification_for_false_keepdims.hpp

```diff
@@ -0,0 +1,155 @@
+    ModifyReduceLogicalOrForFalseKeepDims();
+};
+
+class ov::pass::ModifyReduceForFalseKeepDims : public ov::pass::GraphRewrite {
+public:
+    OPENVINO_RTTI("ModifyReduceForFalseKeepDims", "0");
+    ModifyReduceForFalseKeepDims() {
+        add_matcher<ModifyReduceMeanForFalseKeepDims>();
+        add_matcher<ModifyReduceSumForFalseKeepDims>();
+        add_matcher<ModifyReduceMaxForFalseKeepDims>();
+        add_matcher<ModifyReduceMinForFalseKeepDims>();
+        add_matcher<ModifyReduceLogicalAndForFalseKeepDims>();
+        add_matcher<ModifyReduceLogicalOrForFalseKeepDims>();
+    }
+};
+
+template <class T>
+ov::matcher_pass_callback ModifyReduceBase::modify_reduce_for_false_keepdims() {
+    return [&](ov::pass::pattern::Matcher& m) {
+        auto reduce = std::dynamic_pointer_cast<T>(m.get_match_root());
+        if (!reduce)
+            return false;
+
+        auto input = reduce->input_value(0);
+        const auto input_shape = input.get_shape();
+        const auto reduce_shape = reduce->output(0).get_shape();
+        auto axes_node = std::dynamic_pointer_cast<ov::opset1::Constant>(reduce->input_value(1).get_node_shared_ptr());
+        if (!axes_node) {
+            return false;
+        }
+
+        auto axes_vector = axes_node->template cast_vector<int64_t>();
+        const auto input_rank = input.get_partial_shape().rank().get_length();
+        // Transform negative axes into non-negative ones
+        for (size_t i = 0; i < axes_vector.size(); ++i) {
+            if (axes_vector[i] < 0) {
+                axes_vector[i] += input_rank;
+            }
+        }
+        std::sort(axes_vector.begin(), axes_vector.end());
+
+        if (is_unsupported_reordered_axes(axes_vector, input_rank, (input_rank - 2)) && input_shape.size() < 6) {
+            ngraph::NodeVector new_ops;
+
+            // Reduce
+            auto reduce_const =
+                ov::opset1::Constant::create(ov::element::i64, ov::Shape{axes_vector.size()}, axes_vector);
+            input = std::make_shared<T>(input, reduce_const, true);
+            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name());
+            new_ops.push_back(input.get_node_shared_ptr());
+
+            // Reshape
+            auto reshape_shape = ov::Shape(input_shape.size(), 1);
+            // Expected that a feature axis is only un-reduced.
+            reshape_shape[0] = reduce_shape[0];
+            input = std::make_shared<ov::opset1::Reshape>(
+                input,
+                ov::opset1::Constant::create(ov::element::i64, ov::Shape{reshape_shape.size()}, reshape_shape),
+                false);
+
+            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name() + "_reshape_false_keepdims");
+            new_ops.push_back(input.get_node_shared_ptr());
+
+            copy_runtime_info(reduce, new_ops);
+            reduce->output(0).replace(input);
+        }
+
+        return false;
+    };
+}
+
+namespace ngraph {
```

Comment:
we don't need these aliases for new transformations, please delete them

Reply:
Removed